neuland/digitizing/DigitizingChannel.cxx

```diff
@@ -0,0 +1,54 @@
+/******************************************************************************
+ *   Copyright (C) 2019 GSI Helmholtzzentrum für Schwerionenforschung GmbH    *
+ *   Copyright (C) 2019 Members of R3B Collaboration                          *
+ *                                                                            *
+ *             This software is distributed under the terms of the            *
+ *                 GNU General Public Licence (GPL) version 3,                *
+ *                    copied verbatim in the file "LICENSE".                  *
+ *                                                                            *
+ * In applying this license GSI does not waive the privileges and immunities  *
+ * granted to it by virtue of its status as an Intergovernmental Organization *
+ * or submit itself to any jurisdiction.                                      *
+ ******************************************************************************/
+
+#include "DigitizingChannel.h"
+#include <algorithm>
+
+namespace Digitizing
+{
+    Channel::Channel(ChannelSide side)
+        : fSide(side)
+    {
+    }
+
+    auto Channel::GetRandom3Ref() -> TRandom3&
+    {
+        static auto random3 = TRandom3{};
+        return random3;
+    }
+
+    auto Channel::GetSignals() -> const Signals&
+    {
+        if (!fSignals.valid())
+        {
+            fSignals.set(ConstructSignals());
+        }
+        return fSignals.getRef();
+    }
+
+    auto Channel::HasFired() -> bool { return (!GetSignals().empty()); }
+
+    auto Channel::GetTrigTime() -> double
+    {
+        if (!fTrigTime.valid())
+        {
+            const auto& signals = GetSignals();
```

Comment:
Sometimes is dificult to understand the code because one cannot identify the objects created with "auto", what kind of object is GetSignals()?
const Foo& signals = GetSignals();
will be better.

Reply:
One can easily identify the type with the help from some static analysis tools.

Reply:
Could you provide some examples?